include/triton/Dialect/TritonNvidiaGPU/Transforms/TMAUtilities.h

```diff
@@ -22,30 +111,44 @@ mlir::LogicalResult createTMADesc(mlir::Value tmaPtr,
 
   auto elemType = op.getBase().getType().getPointeeType();
   auto elemSize = elemType.getIntOrFloatBitWidth() / 8;
+  auto mmaEncoding = llvm::dyn_cast_or_null<gpu::NVMMASharedEncodingAttr>(
+      op.getType().getBlockType().getEncoding());
+  bool fp4Padded = mmaEncoding && mmaEncoding.getFp4Padded();
 
   int32_t contig_dim_size = op.getTensorShape().back();
   int32_t contig_dim_size_in_bytes = contig_dim_size * elemSize;
   if (contig_dim_size_in_bytes > 128) {
     contig_dim_size = 128 / elemSize;
   }
   llvm::SmallVector<Value> boxDim;
+  if (fp4Padded && contig_dim_size != 128) {
+    op->emitError(
+        "FP4 padded loads require 128 elements or more in the last dim");
```

Comment:
@masahi I wanted to confirm, previously your code overwrote the box size to be 128 even for smaller sizes. In my PR this was giving illegal instruction errors when run under `cuda-gdb`, but does this happening on your PR as well or is there something this PR is missing?

Reply:
See https://docs.nvidia.com/cuda/parallel-thread-execution/#data-movement-and-conversion-instructions-tensor-copy. `b4x16_p64` requires both the box size and the size of the tensor in smem to be 128. 

"64B" in the doc might be confusing, but what we call "box size" or "tensor dim" are actually "the number of elements of the specified type" - so we have 128 fp4 elements packed in 64B in gmem, for example. That's what the doc means by "Box-Size[0] must be 64B".